pkg/project/v2/project_loader.go

```diff
@@ -84,13 +83,95 @@ func LoadProjects(fs afero.Fs, context ProjectLoaderContext) ([]Project, []error
 		return nil, []error{fmt.Errorf("no projects defined in manifest")}
 	}
 
-	log.Info("Loading %d projects...", len(context.Manifest.Projects))
+	if len(specificProjects) == 0 {
```

Comment:
nit: I think you can let "filterProjectDefinitionsByProjectNames()" just return the "unfiltered" projects if an empty slice is passed.
Then you don't specifically need to check for that case here.

Reply:
>  I feel that not adding special behavior avoids the need for any documentation as the function only does what the name implies.

the name of the function does not really imply what it does with empty slices.

Reply:
The program flow would also be easier too, as there is then really no need to check what function would be executed if specificProjects is empty.... But it's a nit, feel free to ignore :)